presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -213,6 +214,20 @@ public ExpressionAnalyzer(
             Map<NodeRef<Parameter>, Expression> parameters,
             WarningCollector warningCollector,
             boolean isDescribe)
+    {
+        this(metadata, accessControl, statementAnalyzerFactory, session, symbolTypes, parameters, warningCollector, isDescribe, Correlation.ALLOWED);
+    }
+
+    public ExpressionAnalyzer(
+            Metadata metadata,
+            AccessControl accessControl,
+            Function<Node, StatementAnalyzer> statementAnalyzerFactory,
+            Session session,
+            TypeProvider symbolTypes,
+            Map<NodeRef<Parameter>, Expression> parameters,
+            WarningCollector warningCollector,
+            boolean isDescribe,
+            Correlation correlation)
```

Comment:
"correlation" sounds like an actual occurances of a correlation (like correlated identifier(s))
it's not meaningful.

if we cannot find a better name, `boolean correlationAllowed` would be better

Reply:
Renamed to correlationSupport. I chose not to use a boolean because it's easy to confuse with `isDescribe` above and very error prone.